src/shared/signatures.rs

```diff
@@ -62,8 +62,8 @@ impl FunctionInfo {
         FunctionInfo {
             name,
             package_name,
-            required_args: get_argument_names(f.req.clone()),
-            optional_args: get_argument_names(f.opt.clone()),
+            required_args: get_argument_names(&f.req),
+            optional_args: get_argument_names(&f.opt),
```

Comment:
I suspect there is so much low hanging fruit like this if we really sit down and look for it.